easybuild/framework/easyconfig/format/yeb.py

```diff
@@ -141,3 +141,18 @@ def is_yeb_format(filename, rawcontent):
             if line.startswith('name: '):
                 isyeb = True
     return isyeb
+
+def quote_yaml_special_chars(val):
+    """
+    Single-quote values that contain special characters, specifically to be used in YAML context (.yeb files)
+    """
+    if isinstance(val, basestring):
+        if "'" in val:
+            if '"' in val:
+                val = val.replace('"', '\\"')
+            val = '"%s"' % val
+
+        elif YAML_SPECIAL_CHARS.intersection(val):
+            val = "'%s'" % val
```

Comment:
YAML special chars are escaped by quoting the string (as YAML doesn't usually require quoting for regular strings). If the string contains both `'` and any other special char, the string will be quoted with `".."` anyway, so special chars don't matter anymore 

now that im typing this, maybe a better solution is to add `'` to YAML_SPECIAL_CHARS and always escape with double quotes